### PR TITLE
add new custom messages for usage with wheel odometry and with pure pursuit controller

### PR DESCRIPTION
**Pull request by nihaal1**
Created new custom messages, trajectory_point.msg and wheel_odometry.msg

### CODE REVIEW

**Review by CarterDiOrio — APPROVED** [review]

**Review by przb — APPROVED** [review]

File: packages/as_msgs/msg/wheel_odometry.msg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Header header
+float32 x
```
Comment (tichakornw) [inline packages/as_msgs/msg/wheel_odometry.msg:2]: I don't quite understand why there is x, y, heading in wheel odometry. Isn't it just the wheel speed?

**Comment by tichakornw** [comment]
Do you want to rename trajectory_point.msg to pose.msg? Essentially, this is just about the pose of the robot, which can be used in other messages (like localization) too.

File: packages/as_msgs/msg/wheel_odometry.msg (hunk 1 above)
Comment (CarterDiOrio) [reply packages/as_msgs/msg/wheel_odometry.msg:2]: I was thinking of wheel odometry as a current pose estimate calculated over time from the wheel speeds. Is the convention to have it just be wheel speed and have the pose estimate done somewhere else?

**Comment by CarterDiOrio** [comment]
I agree with renaming the trajectory_point msg to be a generic pose msg

File: packages/as_msgs/msg/wheel_odometry.msg (hunk 1 above)
Comment (tichakornw) [reply packages/as_msgs/msg/wheel_odometry.msg:2]: Yes, I think they are different things. Now, once you change trajectory_point.msg to pose.msg, you can use pose.msg to publish the pose estimate from wheel speeds (or anything else like SLAM in the future). It means you have the same message type regardless of how you compute the pose. So in the future, if you use another way to compute the pose, you don't have to change your planning/control algorithm at all.

Comment (CarterDiOrio) [reply packages/as_msgs/msg/wheel_odometry.msg:2]: That all makes sense and makes the codebase more modular, will have us get that done

**Pull request merged**